docs/proposals/20250818-multi-az-apiserver-loadbalancer.md

```diff
@@ -0,0 +1,298 @@
+
+## Summary
+Add first-class Multi-AZ support for the Kubernetes control plane LoadBalancer in Cluster API Provider OpenStack (CAPO). The feature reconciles one Octavia LoadBalancer per Availability Zone (AZ), places each VIP in the intended subnet for that AZ via an explicit AZ→Subnet mapping, and by default registers control plane nodes only with the LB in the same AZ. Operators expose the control plane endpoint via external DNS multi-value A records that point at the per-AZ LB IPs. This proposal is additive and backward compatible.
+
+## Motivation
+- Achieve true multi-AZ resilience for the control plane by avoiding a single VIP dependency.
+- Align control plane networking with existing multi-AZ compute placement goals.
+- Provide clear, portable primitives across Octavia providers with native AZ hints and an explicit, unambiguous mapping between AZs and VIP subnets.
+
+## Goals
+- Create and manage one API server LoadBalancer per configured AZ.
+- Support explicit AZ→Subnet mapping only (no positional mapping).
+- Default to same-AZ LB membership for control plane nodes; allow opt-in cross-AZ registration.
+- Keep the API additive with strong validation, clear events and documentation.
+- Preserve user-provided DNS endpoints; DNS record management remains out of scope.
+
+## Non-Goals
+- Managing or provisioning DNS records.
+- Provider-specific topologies such as ACTIVE_STANDBY across fault domains.
+- Service type LoadBalancer for worker Services.
+
+## User Stories
+1) As a platform engineer, I want per-AZ LBs so a full AZ outage leaves the cluster reachable via DNS multi-A records that resolve to the remaining AZs.
+2) As an operator, I want a safe migration path from single-LB clusters to per-AZ LBs without downtime.
+3) As a security-conscious user, I want to restrict VIP access with allowed CIDRs when supported by my Octavia provider.
+
+## Design Overview
+
+### High-level behavior
+- When enabled and configured with an explicit mapping, CAPO reconciles one LoadBalancer per Availability Zone (AZ).
+- VIP placement is controlled only by an explicit mapping list that binds each AZ to a specific subnet on the LB network.
+- Each per-AZ LB is named with an AZ suffix.
+- Control plane nodes are registered as LB members only in their AZ by default; opt-in cross-AZ membership is supported.
+- Operators expose an external DNS name for the control plane endpoint with one A/AAAA record per AZ LB IP.
+
+### Architecture diagram
+```mermaid
+flowchart LR
+  Clients --> DNS[External DNS zone]
+  DNS -->|A record per AZ| LBa[LB az1]
+  DNS -->|A record per AZ| LBb[LB az2]
+  DNS -->|A record per AZ| LBn[LB azN]
+  subgraph OpenStack
+    LBa --> LaL[Listeners] --> Pa[Pools] --> CP1[Control plane nodes in az1]
+    LBb --> LbL[Listeners] --> Pb[Pools] --> CP2[Control plane nodes in az2]
+    LBn --> LnL[Listeners] --> Pn[Pools] --> CPn[Control plane nodes in azN]
+  end
+```
+
+## Integration with External Global Server Load Balancing (GSLB)
```

Comment:
This section about GSLB seems quite redundant to me. It doesn't really have anything to do with CAPO or the implementation of this feature. It just explains how it can fit in the larger ecosystem. I would prefer to reduce this to just a paragraph under the **Motivation** section.